calico_containers/pycalico/ipam.py

```diff
@@ -112,12 +117,17 @@ def unassign_address(self, pool, address):
         Unassign an IP from a pool.
 
         :param IPPool or IPNetwork pool: The pool that the assignment is from.
+        If the pool is None, get the pool from datastore
         :param IPAddress address: The address to unassign.
-
         :return: True if the address was unassigned, false otherwise. An
         exception is thrown for any error conditions.
         :rtype: bool
         """
+        pool = pool or self.get_pool(address)
```

Comment:
Do we need to worry about the assignment being in a different pool since get_pool only returns a single pool and there could potentially be overlapping pools I think (I don't think we police against that at the moment).